src/components/Web3Status/index.tsx

```diff
@@ -141,6 +153,37 @@ function Web3StatusInner() {
 
   const { hasPendingActivity, pendingActivityCount } = usePendingActivity()
 
+  const displayName = useMemo(() => (account ? ENSName || shortenAddress(account) : null), [account, ENSName])
+  const meta = useRef(getConnectionMeta())
+  useEffect(() => {
+    if (displayName) {
+      const meta: ConnectionMeta = {
+        type: connection.type,
+        address: account,
+        ENSName: ENSName ?? undefined,
+      }
+      setConnectionMeta(meta)
+    }
+  }, [ENSName, account, connection.type, displayName])
+
+  const isENSNameLoading = Boolean(meta.current?.address === account && meta.current?.ENSName && ENSLoading)
```

Comment:
this variable name is also confusing to me, because it includes more than just the loading state. 

is `meta` just supposed to be the cached wallet information? if you can clarify that with a couple comments or name changes then i think this file will be alot easier to follow